### PR TITLE
Tickets/dm 31930

### PR DESCRIPTION
**Pull request by tribeiro**

### CODE REVIEW

**Review by yoachim — CHANGES_REQUESTED** [review]

File: rubin_sim/scheduler/basis_functions/basis_functions.py [hunk 1]
```diff
@@ -1476,3 +1476,62 @@ def _calc_value(self, conditions, **kwargs):
         result[good_pix] = 1
 
         return result
+
+
+class VisitGap(Base_basis_function):
+    def __init__(self, note, gap_min=25.0, penalty_val=np.nan):
```
[the remark this answers is not in the document]
Comment (rhiannonlynne) [reply rubin_sim/scheduler/basis_functions/basis_functions.py:1493]: Question: How is this different from either Avoid_Fast_Revisits or Visit_Repeat_Basis_function (with the max gap set to infinity)? 
I see that here there is no spatial dependence, so it's basically saying "should I do X program or not" (assuming the use of  'note' here is indicating something about the survey itself), which is a bit different maybe? Is that what is driving the addition of another basis function devoted to time between visits? 
Or is the use of 'note' different?

File: rubin_sim/scheduler/surveys/field_survey.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
+import numpy as np
+
+from rubin_sim.utils import _raDec2Hpid
+
+from .dd_surveys import Deep_drilling_survey
```
Comment (rhiannonlynne) [inline rubin_sim/scheduler/surveys/field_survey.py:5]: Seems like we should probably make a more general "user defined scripted survey" class to inherit from, then make that available for this and for DD surveys, and potentially also ToO and star field kind of surveys. Could make a meta-list of what kinds of things people might generally need to customize (filter sequence, locations of fields, numbers of visits and exposure times per visit (per filter perhaps, could expand this to a list matching the filter sequence), basis functions and detailers, other info needed for programming sequences (filter change time, read time, etc) and then let that be the framework for further customization?

File: rubin_sim/scheduler/features/features.py [hunk 3]
```diff
@@ -392,6 +392,20 @@ def add_observation(self, observation, indx=None):
             self.feature[indx] = observation['mjd']
 
 
+class NoteLastObserved(BaseSurveyFeature):
```
[the remark this answers is not in the document]
Comment (tribeiro) [reply rubin_sim/scheduler/features/features.py:395]: Oh! the docstring is very confusing indeed.. Let me work on it...

[the remark this answers is not in the document]
Comment (tribeiro) [reply rubin_sim/scheduler/features/features.py:395]: But this basically keeps track of the last time an observation with a note field that matches it's internal `note` was made...

File: rubin_sim/scheduler/surveys/field_survey.py (hunk 2 above)
Comment (tribeiro) [reply rubin_sim/scheduler/surveys/field_survey.py:5]: I think that is a good idea..

**Pull request merged**